src/index.ts

```diff
@@ -50,8 +50,28 @@ export function isolate<State extends object, Scope extends Lens<State, any> | k
     }
 }
 
+/**
+ * Form component constructor
+ * @param fields - Field implementations for the given `FormDeclaration`.
+ * @param options - Options to customize form behavior. `customSubmission` enable the form to be submitted by custom keybinds like 'Ctrl + Enter'. `predicate` is a function which defines what custom keybinds are. By default, 'Ctrl + Enter' and 'Metakey + Enter' are treated as submission keybinds. `fields` is the set of the form field which accepts custom keybinds.
+ *
+ * @example
+ * form(fieldsConstructor(), {
+ *   customSubmission: {
+ *     // We can submit this form by pressing 'Ctrl + Enter' of 'Metakey + Enter' while editing 'description' fields
+ *     fields: ['description'],
+ *   },
+ * })
+ *
+ */
 export function form<Decl extends FormDeclaration<any>>(
     fields: FieldsFor<Decl>,
+    options: {
```

Comment:
Let's define these types separately and export them.

```typescript
export type Options = Partial<Readonly<{
    customSubmission: Options.CustomSubmission;
}>>

export namespace Options {
    export type CustomSubmission<Decl extends FormDeclaration<any>> = Readonly<{
        fields: Set<keyof Decl>;
        predicate(e: KeyboardEvent): boolean;
    }>;
}
```

```suggestion
    options: Options
```

Reply:
https://github.com/herp-inc/cyclic-form/pull/2/commits/3d2a5314797f39f96afa32fd4f52efa1c56a3d1a
I think that `Options` does not have to be `Partial<Readonly<{ customSubmission: ... }>>` because we use default parameter in `form`.